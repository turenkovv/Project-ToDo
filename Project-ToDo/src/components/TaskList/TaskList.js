import React from 'react';
import Task from './Task/Task';
import { connect } from 'react-redux';


const TaskList = ({tasks}) => {
    return (
        <div style={{marginTop:"11px", width: "100%", display: "flex", flexDirection: "column", alignItems: "center"}}>
            {tasks.map(({ id, name, description, status }) => <Task id={id} name={name} description={description} status={status} key={id} />)}
        </div>
    );
};


const mapStateToProps = (state) => {
    return{
        tasks: state.todo.todoList
    };
};


export default connect(mapStateToProps)(TaskList);