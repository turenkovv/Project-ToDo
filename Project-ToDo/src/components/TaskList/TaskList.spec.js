import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'; // підключимо redux-mock-store
import TaskList from './TaskList';

// Створимо фейковий store за допомогою redux-mock-store
const mockStore = configureStore([]);

describe('TaskList component', () => {
  test('renders correctly', () => {
    // Створимо фейковий стан для store
    const initialState = {
      todo: {
        todoList: [
          { id: 1, name: 'Task 1', description: 'Description 1', status: false },
          { id: 2, name: 'Task 2', description: 'Description 2', status: true },
        ],
      },
    };
    const store = mockStore(initialState);

    const component = renderer.create(
      <Provider store={store}>
        <TaskList />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
