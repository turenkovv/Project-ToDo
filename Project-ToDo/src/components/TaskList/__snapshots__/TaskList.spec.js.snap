// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList component renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "marginTop": "11px",
      "width": "100%",
    }
  }
>
  <div
    className="task"
  >
    <label
      className="checkbox"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkmark"
      />
    </label>
    <div
      className="text name"
    >
      Task 1
    </div>
    <div
      className="text desc"
    >
      Description 1
    </div>
    <button
      className="deleteBtn"
      onClick={[Function]}
    >
      <img
        alt=""
        src="assets/bin.png"
      />
    </button>
  </div>
  <div
    className="task"
  >
    <label
      className="checkbox"
    >
      <input
        checked={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="checkmark"
      />
    </label>
    <div
      className="text name"
    >
      Task 2
    </div>
    <div
      className="text desc"
    >
      Description 2
    </div>
    <button
      className="deleteBtn"
      onClick={[Function]}
    >
      <img
        alt=""
        src="assets/bin.png"
      />
    </button>
  </div>
</div>
`;
